src/subgraph.js

```diff
@@ -317,6 +318,76 @@ ${abiFunctions
       }, immutable.List())
   }
 
+  static validateBlockFunctions(manifest, { resolveFile }) {
```

Comment:
I forgot how hard it is to review JS code, thanks for the reminder.

Does dataSource really have a `.getIn` method? Sure! One must assume.

Also, it's interesting to see what looks like a mature and fully-fledged persistent immutable data structure library in JS.

Reply:
Yeah, `immutable.js` is rarely seen in the wild but it is mature. Unfortunately, it's rather unidiomatic, hence a lot of graph-cli is a little... odd.